File: packages/point/src/std/yaml.ts
import { parse as parseYaml, stringify as stringifyYaml } from "yaml";

export type PointStdError = { message: string };

export function yamlParse(value: string): string | PointStdError {
	try {
		return JSON.stringify(parseYaml(value));
	} catch (error) {
		return { message: error instanceof Error ? error.message : String(error) };
	}
}

export function yamlStringify(value: string): string {
	return stringifyYaml(JSON.parse(value));
}

File: packages/point/python_std/point_std/yaml.py
from __future__ import annotations

import json
from typing import Any


def _compact_json(value: Any) -> str:
	return json.dumps(value, separators=(",", ":"), ensure_ascii=False)


def yamlParse(value: str) -> str | dict[str, str]:
	try:
		import yaml
	except ImportError:
		return {"message": "PyYAML is required for std.yaml in Python emit"}
	try:
		parsed = yaml.safe_load(value)
		return _compact_json(parsed)
	except Exception as error:
		return {"message": str(error)}


def yamlStringify(value: str) -> str:
	import yaml

	return yaml.safe_dump(json.loads(value), sort_keys=False)
